test/com/google/enterprise/adaptor/ad/AdAdaptorTest.java

```diff
@@ -1425,6 +1419,44 @@ public void testFakeAdaptorGetDocIds() throws Exception {
   }
 
   @Test
+  public void testMultipleFullGroupPushes() throws Exception {
+    MockLdapContext ldapContext = defaultMockLdapContext();
+    AdAdaptor adAdaptor = new FakeAdaptor(ldapContext);
+    RecordingDocIdPusher pusher = new RecordingDocIdPusher();
+    Map<String, String> configEntries = defaultConfig();
+    pushGroupDefinitions(adAdaptor, configEntries, pusher, /*fullPush=*/ true,
```

Comment:
I find this (existing) helper method to be marginal at best, and obfuscatory when init is false.

Reply:
This started out as a copy/paste of the previous test and only realized later that the helper method barely helped. I could do a separate sweep to eliminate its use.